### PR TITLE
CDSv2 fixes, and parity with v1alpha3

### PR DESCRIPTION
**Pull request by rshriram**
Fix incorrect cluster generation for CDS. Also add support for configuring LB policies, TLS settings using DestinationRules.
Set the H2 upgrade to use the downstream protocol for the moment.

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: pilot/pkg/networking/v1alpha3/cluster.go [hunk 1]
```diff
@@ -44,14 +56,15 @@ func BuildClusters(env model.Environment, proxy model.Proxy) []*v2.Cluster {
 			return nil
 		}
 
-		clusters = append(clusters, buildInboundClusters(instances)...)
+		clusters = append(clusters, buildInboundClusters(env, instances)...)
 
+		// TODO: Bug? why only for sidecars?
```
Comment (ijsnellf) [inline pilot/pkg/networking/v1alpha3/cluster.go]: routers won't have co-located instances.

File: pilot/pkg/networking/v1alpha3/cluster.go [hunk 2]
```diff
@@ -60,36 +73,38 @@ func buildOutboundClusters(env model.Environment, services []*model.Service) []*
...
+		for _, port := range service.Ports {
+			hosts := buildClusterHosts(env, service, port)
+
+			// create default cluster
+			defaultCluster := &v2.Cluster{
+				Name:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port),
+				Type:  convertResolution(service.Resolution),
+				Hosts: hosts,
+			}
+			defaultCluster.ProtocolSelection = v2.Cluster_USE_DOWNSTREAM_PROTOCOL
```
Comment (ijsnellf) [inline pilot/pkg/networking/v1alpha3/cluster.go]: nit: set the field
`ProtocolSelection: v2.Cluster_USE_DOWNSTREAM_PROTOCOL`

File: pilot/pkg/networking/v1alpha3/cluster.go [hunk 3]
```diff
@@ -60,36 +73,38 @@ func buildOutboundClusters(env model.Environment, services []*model.Service) []*
...
 						Type:  convertResolution(service.Resolution),
 						Hosts: hosts,
 					}
-					applyTrafficPolicy(cluster, destinationRule.TrafficPolicy)
-					applyTrafficPolicy(cluster, subset.TrafficPolicy)
-
-					clusters = append(clusters, cluster)
+					applyTrafficPolicy(subsetCluster, destinationRule.TrafficPolicy)
+					applyTrafficPolicy(subsetCluster, subset.TrafficPolicy)
+					subsetCluster.ProtocolSelection = v2.Cluster_USE_DOWNSTREAM_PROTOCOL
```
Comment (ijsnellf) [inline pilot/pkg/networking/v1alpha3/cluster.go]: see above

File: pilot/pkg/networking/v1alpha3/cluster.go (hunk 1 above)
Comment (rshriram) [reply pilot/pkg/networking/v1alpha3/cluster.go]: No.. this is for the JWKURI thing.. not for proxying

**Pull request merged**